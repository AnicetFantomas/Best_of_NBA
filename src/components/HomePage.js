import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import userIcon from './Images/user.png';
import './Styles/Home.css';
import arrow from './Images/arrow.png';
import searchIcon from './Images/search.png';
import { fetchCryptoDetails } from '../Redux/FetchDtails';

const HomePage = () => {
  const getAllCryptos = useSelector((state) => state.cryptos);
  const dispatch = useDispatch();

  const [initialSearch, setSearch] = useState('');
  const matchedCryptos = getAllCryptos.filter((item) => item.cryptoSymb
    .toLowerCase().includes(initialSearch.toLowerCase()));

  const handleChange = (e) => {
    setSearch(e.target.value);
  };

  useEffect(() => {
    if (!getAllCryptos.length) dispatch(fetchCryptoDetails());
  }, [dispatch]);

  return (
    <>
      <div className="logo-container">
        <div className="logo">
          Your
          <span>Crypto</span>
          Day
        </div>
        <div><img className="user-icon" src={userIcon} alt="user-icon" /></div>
      </div>
      <div className="header">
        <div className="header-text">
          <p>Start The day with coffee,</p>
          <p>End The day with crypto</p>
        </div>
      </div>
      <div>
        <div className="search-container">
          <p className="top-cryptos">Top 50 current cryptos</p>
          <div className="search-box">
            <input
              type="search"
              placeholder="search..."
              onChange={handleChange}
              value={initialSearch}
              className="input"
            />
            <div className="search-img-cont"><img className="search-icon" src={searchIcon} alt="search-icon" /></div>
          </div>
        </div>
      </div>
      <div className="samples-container">
        {matchedCryptos.slice(0, 51).map((crypto) => (
          <Link to={`crypto/${crypto.cryptoSymb}`} className="sample" key={crypto.cryptoSymb}>
            <div><img src={arrow} className="image" alt="arrow" /></div>
            <div className="text-container">
              <div className="symbole">{crypto.cryptoSymb}</div>
              <span>
                {crypto.priceChangePercent}
                %
              </span>
              <p>{crypto.prevClosePrice}</p>
            </div>
          </Link>
        ))}
      </div>
    </>
  );
};

export default HomePage;
