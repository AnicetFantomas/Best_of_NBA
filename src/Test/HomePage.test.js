import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../components/configuration';
import HomePage from '../components/HomePage';

describe('missions Component test', () => {
  it('Renders Mission Row', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <HomePage />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
