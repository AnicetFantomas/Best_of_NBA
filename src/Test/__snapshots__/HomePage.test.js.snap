// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missions Component test Renders Mission Row 1`] = `
Array [
  <div
    className="logo-container disp-flex"
  >
    <div
      className="logo asef-center"
    >
      Your
      <span>
        Crypto
      </span>
      Day
    </div>
    <div>
      <img
        alt="user-icon"
        className="user-icon asef-center"
        src="user.png"
      />
    </div>
  </div>,
  <div
    className="header"
  >
    <div
      className="header-text asef-center"
    >
      <p>
        Start The day with coffee,
      </p>
      <p>
        End The day with crypto
      </p>
    </div>
  </div>,
  <div>
    <div
      className="search-container disp-flex"
    >
      <p
        className="top-cryptos"
      >
        Top 50 current cryptos
      </p>
      <div
        className="search-box disp-flex"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="search..."
          type="search"
          value=""
        />
        <div
          className="search-img-cont disp-flex"
        >
          <img
            alt="search-icon"
            className="search-icon asef-center"
            src="search.png"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="samples-container"
  />,
]
`;
